components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/jwt_auth_provider.bal

```diff
@@ -53,14 +54,20 @@ public type JwtAuthProvider object {
                 setErrorMessageToInvocationContext(API_AUTH_INVALID_CREDENTIALS);
                 return handleVar;
             }
-
             boolean isBlacklisted = false;
             string? jti = "";
             runtime:InvocationContext invocationContext = runtime:getInvocationContext();
             runtime:AuthenticationContext? authContext = invocationContext?.authenticationContext;
             if (authContext is runtime:AuthenticationContext) {
                 string? jwtToken = authContext?.authToken;
                 if (jwtToken is string) {
+                    (jwt:JwtPayload | error) payload = getDecodedJWTPayload(jwtToken);
```

Comment:
since you get the decoded payload here, shall we reuse it 74th line, without getting the token from the cache again.